### PR TITLE
in isValidAddress check that address starting with 'r'

### PR DESCRIPTION
**Pull request by darkdarkdragon**
because UInt160 considers valid hex values

### CODE REVIEW

File: src/api/common/schema-validator.js [hunk 1]
```diff
@@ -12,7 +12,8 @@ const ValidationError = require('./errors').ValidationError;
 let SCHEMAS = {};
 
 function isValidAddress(address: string): boolean {
-  return core.UInt160.is_valid(address);
+  return typeof address === 'string' && address.length && address[0] === 'r' &&
```
Comment (sublimator) [inline src/api/common/schema-validator.js]: Flow seems to be complaining about:
```
>>> true && 0 && false
0
```

You might need to wrap the return in Boolean to shut it up?

Comment (darkdarkdragon) [inline src/api/common/schema-validator.js]: Just `length > 0` is shorter :smile:

**Comment by wltsmrz** [comment]
LGTM

File: src/api/common/schema-validator.js [hunk 2]
```diff
@@ -12,7 +12,9 @@ const ValidationError = require('./errors').ValidationError;
 let SCHEMAS = {};
 
 function isValidAddress(address: string): boolean {
-  return core.UInt160.is_valid(address);
+  return typeof address === 'string' && address.length > 0 &&
```
Comment (clark800) [inline src/api/common/schema-validator.js]: `typeof address === 'string'` is redundant

Comment (darkdarkdragon) [inline src/api/common/schema-validator.js]: This function is exposed as public api, so it can be called with any value.

Comment (clark800) [inline src/api/common/schema-validator.js]: Ah good point, flow doesn't help if client code isn't using it!

**Comment by clark800** [comment]
LGTM

File: src/api/common/schema-validator.js (hunk 2 above)
Comment (darkdarkdragon) [inline src/api/common/schema-validator.js]: And in most cases it will not :smile:

**Pull request merged**